newIDE/app/src/ResourcesList/index.js

```diff
@@ -77,29 +81,24 @@ export default class ResourcesList extends React.Component<Props, State> {
     this.props.onDeleteResource(resource);
   };
 
-  _scanForNewResources = () => {
+  _scanForNewResources = (extensions: string, resource: gdResource) => {
```

Comment:
You're not using resource in the function, I think you forgot it.
Also, what you're receiving should not be a gdResource, but a constructor (on which you can do `new Resource()`);

It may be a bit tricky to properly flow type, I think we're better with a "creator" function:
```js
_scanForNewResources = (extensions: string, createResource: () => gdResource) => {
...
createOrUpdateResource(project, createResource(), fileName);
```